beacon-chain/state/v2/state_trie.go

```diff
@@ -35,6 +37,27 @@ func InitializeFromProto(st *ethpb.BeaconStateAltair) (*BeaconState, error) {
 	return InitializeFromProtoUnsafe(proto.Clone(st).(*ethpb.BeaconStateAltair))
 }
 
+// InitializeFromSSZReader can be used when the source for a serialized BeaconState object
+// is an io.Reader. This allows client code to remain agnostic about whether the data comes
+// from the network or a file without needing to read the entire state into mem as a large byte slice.
+func InitializeFromSSZReader(r io.Reader) (*BeaconState, error) {
```

Comment:
Curious how we can make this more forward-compatible say when `BeaconStateMerge` comes around. Do we duplicate code into `v3` pkg?

Reply:
Everyone landed on the same (totally valid) concern haha, copying one more time ;)
Working on making this more future proof in cp-sync-sniff-origin-init 65912a9
Still need to do a bit of testing but i think that work will decouple this code from relying on a particular proto struct.